client/src/hooks/Input/useTextarea.ts

```diff
@@ -158,26 +148,27 @@ export default function useTextarea({
     getSender,
     assistantName,
     textAreaRef,
-    assistantMap,
   ]);
 
   const handleKeyDown = (e: KeyEvent) => {
     if (e.key === 'Enter' && isSubmitting) {
       return;
     }
 
-    const isNonShiftEnter = e.key === 'Enter' && !e.shiftKey;
-
-    if (isNonShiftEnter && filesLoading) {
-      e.preventDefault();
-    }
-
-    if (isNonShiftEnter) {
-      e.preventDefault();
-    }
-
-    if (isNonShiftEnter && !isComposing?.current) {
-      submitButtonRef.current?.click();
+    if (e.key === 'Enter' && !enterToSend) {
+      if (!e.shiftKey) {
+        e.preventDefault();
+        if (textAreaRef.current && textAreaRef.current.value !== null) {
+          insertTextAtCursor(textAreaRef.current, '\n');
```

Comment:
not sure why this is necessary, seems to work without `insertTextAtCursor`?

Reply:
solved  2832434ba157810b771043d729b5ff60346ce278